pkg/raft/raft.go

```diff
@@ -968,22 +968,13 @@ func (r *raft) appliedSnap(snap *pb.Snapshot) {
 // only be called in StateLeader.
 func (r *raft) maybeCommit() bool {
 	index := r.trk.Committed()
-	if index <= r.raftLog.committed {
-		// The commit index must not regress.
-		// TODO(pav-kv): consider making it an assertion.
```

Comment:
NB: We can't make it an assertion because this inequality can happen, so I removed this TODO.

The reason why it can happen is that, initially, the leader doesn't know up-to-date `Match` indices for the followers. So, initially, `r.trk.Committed()` will return stale indices that may be behind `raftLog.committed`.

Could add a comment to this extent.

Reply:
I'd appreciate a comment.